### PR TITLE
ZEN-28005 XSS fix

### PR DESCRIPTION
**Pull request by mayathegreat**
initial commit

### CODE REVIEW

File: Products/Zuul/routers/zep.py [hunk 1]
```diff
@@ -919,6 +919,10 @@ def add_event(self, summary, device, component, severity, evclasskey,
         """
         device = device.strip()  # ZEN-2479: support entries like "localhost "
         try:
+            # Fix for ZEN-28005 to thwart XSS attacks from incoming events
```
Comment (jplouis) [inline Products/Zuul/routers/zep.py]: Any reason not to escape the same fields from above?

File: Products/ZenHub/XmlRpcService.py [hunk 2]
```diff
@@ -36,6 +37,11 @@ def __init__(self, dmd):
 
 
     def xmlrpc_sendEvent(self, data):
+        # Fix for ZEN-28005 to thwart XSS attacks from incoming events
+        for f in data:
+            if f in ("eventClassKey", "component", "summary", "device", "eventClass"):
+                if data[f] is not None and len(data[f]) > 0:
+                    data[f] = cgi.escape(data[f])
```
Comment (jplouis) [inline Products/ZenHub/XmlRpcService.py]: Should this be added to zen.sendevent instead to catch other entry points?

File: Products/Zuul/routers/zep.py (hunk 1 above)
Comment (mayathegreat) [inline Products/Zuul/routers/zep.py]: done

File: Products/ZenHub/XmlRpcService.py (hunk 2 above)
Comment (mayathegreat) [inline Products/ZenHub/XmlRpcService.py]: there seems to be a few sendevent methods -- is this the specific one you are referring to? https://github.com/zenoss/zenoss-prodbin/blob/develop/Products/ZenEvents/SendEvent.py

Comment (jplouis) [inline Products/ZenHub/XmlRpcService.py]: I meant this one: https://github.com/zenoss/zenoss-prodbin/blob/2cb7c3de2d33a94d6a4c161817ea0cf68bccf03b/Products/ZenEvents/MySqlSendEvent.py#L56-L56

**Pull request merged**